modules/api/src/main/java/org/apache/ignite/sql/ColumnMetadata.java

```diff
@@ -40,16 +42,71 @@ public interface ColumnMetadata {
 
     /**
      * Returns SQL column type.
-     * TODO: IGNITE-16962 replace return type regarding the SQL type system.
      *
      * @return Value type.
      */
-    Object type();
+    SqlColumnType type();
+
+    /**
+     * Returns SQL column precision.
+     *
+     * @return Value precision.
```

Comment:
it is worth mentioning in the javadoc the value that will be returned for precision/scale in case those characteristics are not applicable